source.go

```diff
@@ -147,6 +147,7 @@ func getPrimaryKey(db *sqlx.DB, database, table string) (common.PrimaryKeyName,
 			constraint_name = 'PRIMARY'
 			AND table_schema = ?
 			AND table_name = ?
+			ORDER BY ORDINAL_POSITION DESC
```

Comment:
Disclaimer: I'm not familiar with how this works in MySQL.:D It's great to have this option that automatically finds a key, but given that it doesn't always work, it's important to make they key column(s) configurable by a user.

Reply:
@nickchomey The best way I can think of implementing this would be using dynamic configuration parameters.

Here's an article explaining a bit how it works: https://meroxa.com/blog/conduit-0.10-comes-with-multiple-collections-support.

Reply:
> a default mechanism (eg use the primary key if there's only one PK column) and then only optionally specify a ID column for particular tables that have composite PKs?

yes, the latter. The connector needs an ordinal column to sort the rows for the snapshot to be restartable. If an ordinal primary key cannot be found, then you would have to specify the column for that table that can be used to sort the rows.

I haven't thought about the correlation with the `tables` parameter being a regex, but for the moment both can work independently, and rely on user discipline to write the correct configuration. Something like:

On snapshot start for table X:

- Look up the ordinal column name in the configuration somehow.
- If found, use that for sorting the rows. Also write that column in the record positions.
- Otherwise, try fin the primary key for the table:
    - if found, do the snapshot.
    - if not found for whatever reason, fail the snapshot and output a descriptive error on what happened in which table.
    
I might implement it with more guardrails to prevent misconfiguration, I'll let you know on the pr, I first have to implement it in order to see the limitations.